Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -1382,23 +1382,24 @@ private static string TimeIndication(ExecutionEngineOptions options, Verificatio
                 if (callError != null)
                 {
                   tok = callError.FailingCall.tok;
-                  msg = callError.FailingCall.ErrorData as string ?? "A precondition for this call might not hold.";
+                  msg = callError.FailingCall.ErrorData as string ??
+                        callError.FailingCall.Description.FailureDescription;
                 }
                 else if (returnError != null)
                 {
                   tok = returnError.FailingReturn.tok;
-                  msg = "A postcondition might not hold on this return path.";
+                  msg = returnError.FailingReturn.Description.FailureDescription;
                 }
                 else
                 {
                   tok = assertError.FailingAssert.tok;
                   if (assertError.FailingAssert is LoopInitAssertCmd)
```

Comment:
Can we simplify this big `if` into something like?
```csharp
if (assertError.FailingAssert.ErrorMessage == null || options.ForceBplErrors)
{
  msg = assertError.FailingAssert.ErrorData as string;
}
else
{
  msg = assertError.FailingAssert.ErrorMessage;
}
msg ??= assertError.FailingAssert.Description.FailureDescription;
```

Reply:
Yeah, definitely.